x-pack/plugins/observability_solution/infra/public/components/asset_details/tabs/overview/services.tsx

```diff
@@ -43,8 +50,26 @@ export const ServicesContent = ({
     }),
     [hostName, dateRange.from, dateRange.to]
   );
-  const { error, loading, response } = useServices(params);
-  const services = response?.services;
+
+  const query = useMemo(() => ({ ...params, filters: JSON.stringify(params.filters) }), [params]);
+
+  const { data, status, error } = useFetcher(
+    async (callApi) => {
+      const response = await callApi('/api/infra/services', {
+        method: 'GET',
+        query,
+      });
+
+      return decodeOrThrow(ServicesAPIResponseRT)(response);
+    },
```

Comment:
Am I wrong or is this pattern repeated all over the place?
```
return decodeOrThrow(SOME_TYPE_RT)(response);
```

Have you considered doing it on the `useFetcher` instead? Just brainstorming.
```suggestion
  const { data, status, error } = useFetcher(
    ServicesAPIResponseRT,
    async (callApi) => {
      return callApi('/api/infra/services', {
        method: 'GET',
        query,
      });
    },
```